File: frontend/src/types/operation-log.ts
import type { PageQuery } from './common'

export interface OperationLog {
  id: number
  operator_id?: number
  operator_name?: string
  change_summary: string
  module: string
  action: string
  object_type: string
  object_id: string
  before_data?: Record<string, unknown>
  after_data?: Record<string, unknown>
  reason?: string
  ip_address?: string
  created_at: string
}

export interface OperationLogQuery extends PageQuery {
  operator_id?: number
  module?: string
  start_date?: string
  end_date?: string
}

File: backend/app/schemas/operation_log.py
from datetime import datetime
from typing import Any

from app.schemas.common import ORMModel


class OperationLogResponse(ORMModel):
    id: int
    operator_id: int | None
    operator_name: str | None = None
    change_summary: str
    module: str
    action: str
    object_type: str
    object_id: str
    before_data: dict[str, Any] | None
    after_data: dict[str, Any] | None
    reason: str | None
    ip_address: str | None
    created_at: datetime
